string key
uint8 valuetype
string value